prediction-insights.ipynb

```diff
@@ -0,0 +1,774 @@
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "   sepal-length  sepal-width  petal-length  petal-width  target  \\\n",
+       "4           4.5          2.3           1.3          0.3       0   \n",
+       "1           5.0          2.0           3.5          1.0       1   \n",
+       "3           5.1          2.5           3.0          1.1       1   \n",
+       "5           5.2          2.7           3.9          1.4       1   \n",
+       "0           4.9          2.4           3.3          1.0       1   \n",
+       "2           5.0          2.3           3.3          1.0       1   \n",
+       "6           5.5          2.4           3.7          1.0       1   \n",
+       "7           5.5          2.4           3.8          1.1       1   \n",
+       "\n",
+       "   similarity_conviction  \n",
+       "4               0.243127  \n",
+       "1               0.572244  \n",
+       "3               0.651609  \n",
+       "5               0.715024  \n",
+       "0               0.875392  \n",
+       "2               0.938972  \n",
+       "6               1.123037  \n",
+       "7               1.326493  "
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "inf_case_indices = influential_cases[[\".session\", \".session_training_index\"]].values.tolist()\n",
+    "\n",
+    "anom_df = t.get_cases(\n",
+    "    case_indices=inf_case_indices,\n",
+    "    features=[\"sepal-length\", \"sepal-width\", \"petal-length\", \"petal-width\", \"target\", \"similarity_conviction\"]\n",
+    ")\n",
+    "\n",
+    "anom_df.sort_values(by=\"similarity_conviction\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Insight 3: Explanation\n",
+    "\n",
+    "Howso also provides detailed explanations about each prediction. Commonly used explanations are:\n",
+    "\n",
+    "- Feature Contributions for measuring feature importance\n",
+    "- Residuals for measuring the uncertainty of the local data\n",
+    "- Conviction for measurements that are relative to the expected value\n",
+    "\n",
+    "Feature contributions will help determine how much each context feature contributed to the prediction, on the scale\n",
+    "of the action feature."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'petal-width': 0.28871181572340304,\n",
+       " 'petal-length': 0.2931029454393371,\n",
+       " 'sepal-length': 0.04494568151237635,\n",
+       " 'sepal-width': 0.15656876988315238}"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "reaction[\"details\"][\"feature_contributions\"][0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
```

Comment:
why does this matter? can we also try to relate residual to uncertainty. (and maybe even back to the confidence score? idk)